src/IO/JSON/JSONIOHandlerImpl.cpp

```diff
@@ -205,20 +278,23 @@ void JSONIOHandlerImpl::createDataset(
         setAndGetFilePosition(writable, name);
         auto &dset = jsonVal[name];
         dset["datatype"] = datatypeToString(parameter.dtype);
+        auto extent = parameter.extent;
         switch (parameter.dtype)
         {
         case Datatype::CFLOAT:
         case Datatype::CDOUBLE:
         case Datatype::CLONG_DOUBLE: {
-            auto complexExtent = parameter.extent;
-            complexExtent.push_back(2);
-            dset["data"] = initializeNDArray(complexExtent);
+            extent.push_back(2);
             break;
         }
         default:
-            dset["data"] = initializeNDArray(parameter.extent);
             break;
         }
+        // TOML does not support nulls, so initialize with zero
```

Comment:
Interesting discussion: https://github.com/toml-lang/toml/issues/30

Reply:
Agreed, that makes total sense. Should we be a bit more explicit in our guidance in `docs/source/backends/json.rst` to avoid that users mistake it as a full-fledged, high-performance data backend?

Reply:
Great idea about #1493, yes! :)